File: codegen/results/v001/qwen3-coder_30b/20260409_103722/typescript/007_beatles_interview/solution.ts
import fs from 'fs';
import path from 'path';

// Helper function to calculate age as of July 1, 2025
function calculateAge(birthDate: string): number {
  const birth = new Date(birthDate);
  const referenceDate = new Date('2025-07-01');
  
  let age = referenceDate.getFullYear() - birth.getFullYear();
  const monthDiff = referenceDate.getMonth() - birth.getMonth();
  
  if (monthDiff < 0 || (monthDiff === 0 && referenceDate.getDate() < birth.getDate())) {
    age--;
  }
  
  return age;
}

// Read the CSV file
const csvContent = fs.readFileSync(path.join('input', 'input.csv'), 'utf-8');

// Parse CSV lines
const lines = csvContent.trim().split('\n');
const headers = lines[0].split(',').map(header => header.trim());

// Process each data line
const result = lines.slice(1).map(line => {
  const values = line.split(',').map(value => value.trim());
  const person: any = {};
  
  headers.forEach((header, index) => {
    const value = values[index];
    
    // Handle different field types
    if (header === 'birth_date') {
      person[header] = value;
      person.age = calculateAge(value);
    } else if (header === 'salary') {
      person[header] = parseFloat(value);
    } else {
      person[header] = value;
    }
  });
  
  return person;
});

// Output JSON array
console.log(JSON.stringify(result, null, 2));

File: codegen/results/v001/qwen3-coder_30b/20260409_103722/python/007_beatles_interview/solution.py
import csv
import json
from datetime import date

def calculate_age(birth_date_str):
    birth_date = date.fromisoformat(birth_date_str)
    reference_date = date(2025, 7, 1)
    age = reference_date.year - birth_date.year
    if reference_date.month < birth_date.month or (reference_date.month == birth_date.month and reference_date.day < birth_date.day):
        age -= 1
    return age

output = []
with open('input/input.csv', 'r') as file:
    reader = csv.DictReader(file)
    for row in reader:
        output.append({
            'id': int(row['id']),
            'name': row['name'],
            'age': calculate_age(row['birth_date']),
            'email': row['email']
        })

print(json.dumps(output, indent=None))